data-prepper-plugins/aggregate-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/aggregate/AggregateProcessorIT.java

```diff
@@ -235,6 +238,75 @@ void aggregateWithPutAllActionAndCondition() throws InterruptedException {
         }
     }
 
+    @RepeatedTest(value = 2)
+    void aggregateWithRateLimiterAction() throws InterruptedException, NoSuchFieldException, IllegalAccessException {
+        RateLimiterAggregateActionConfig rateLimiterAggregateActionConfig = new RateLimiterAggregateActionConfig();
+        setField(RateLimiterAggregateActionConfig.class, rateLimiterAggregateActionConfig, "eventsPerSecond", 500);
+        aggregateAction = new RateLimiterAggregateAction(rateLimiterAggregateActionConfig);
+        when(pluginFactory.loadPlugin(eq(AggregateAction.class), any(PluginSetting.class)))
+                .thenReturn(aggregateAction);
+        when(aggregateProcessorConfig.getGroupDuration()).thenReturn(Duration.ofSeconds(GROUP_DURATION_FOR_ONLY_SINGLE_CONCLUDE));
+        final AggregateProcessor objectUnderTest = createObjectUnderTest();
+
+        final ExecutorService executorService = Executors.newFixedThreadPool(NUM_THREADS);
+        final CountDownLatch countDownLatch = new CountDownLatch(NUM_THREADS);
+
+        objectUnderTest.doExecute(eventBatch);
+        Thread.sleep(GROUP_DURATION_FOR_ONLY_SINGLE_CONCLUDE * 1000);
+
+        for (int i = 0; i < NUM_THREADS; i++) {
+            executorService.execute(() -> {
+                final List<Record<Event>> recordsOut = (List<Record<Event>>) objectUnderTest.doExecute(eventBatch);
+                for (final Record<Event> record : recordsOut) {
+                    final Map<String, Object> map = record.getData().toMap();
+                    aggregatedResult.add(map);
+                }
+                countDownLatch.countDown();
+            });
+        }
+
+        boolean allThreadsFinished = countDownLatch.await(5L, TimeUnit.SECONDS);
+
+        assertThat(allThreadsFinished, equalTo(true));
+        // Expect less number of events to be received, because of rate limiting
+        assertThat(aggregatedResult.size(), lessThan(NUM_THREADS * NUM_EVENTS_PER_BATCH));
+    }
```

Comment:
This makes sense, result size should be less than total events sent due to rate limiting. 
But on the other hand, the result size should also be less than or equal to (rate * time period) -- `eventsPerSecond * GROUP_DURATION_FOR_ONLY_SINGLE_CONCLUDE`. Is that right?